custom_components/adaptive_lighting/switch.py

```diff
@@ -1717,21 +1778,29 @@ async def state_changed_event_listener(self, event: Event) -> None:
             # incorrect 'min_kelvin' and 'max_kelvin', which happens e.g., for
             # Philips Hue White GU10 Bluetooth lights).
             old_state: list[State] | None = self.last_state_change.get(entity_id)
-            if (
-                old_state is not None
-                and old_state[0].context.id == new_state.context.id
-            ):
-                # If there is already a state change event from this event (with this
-                # context) then append it to the already existing list.
-                _LOGGER.debug(
-                    "State change event of '%s' is already in 'self.last_state_change' (%s)"
-                    " adding this state also",
-                    entity_id,
-                    new_state.context.id,
-                )
+            if is_our_context(new_state.context):
+                if (
+                    old_state is not None
+                    and old_state[0].context.id == new_state.context.id
+                ):
+                    _LOGGER.debug(
+                        "TurnOnOffListener: State change event of '%s' is already"
+                        " in 'self.last_state_change' (%s)"
+                        " adding this state also",
+                        entity_id,
+                        new_state.context.id,
+                    )
+                    self.last_state_change[entity_id].append(new_state)
+                else:
+                    _LOGGER.debug(
+                        "TurnOnOffListener: New adapt '%s' found for %s",
+                        new_state,
+                        entity_id,
+                    )
+                    self.last_state_change[entity_id] = [new_state]
+                    self.start_transition_timer(entity_id)
```

Comment:
Here we are calling:
```
if is_our_context(new_state.context):
    ...
    self.start_transition_timer(entity_id)
```
so that means if an external party (e.g., Google Home) requested a light change, `detect_non_ha_changes` would never work correctly.

Reply:
@basnijholt None of what the listener is doing is relevant to `significant_change` anymore. Please take a closer look at that function.

The listener is only relevant for ensuring an adapt was correctly received by the bulb so we can start the transition timer.

Reply:
I've already verified `significant_change` fires manual control events when changed with my non ha dimmer.

most of the listener code you posted here is only relevant to wait for transition stuff and the `alt_detect_method` pr.

Reply:
`last_state_change` isn't involved **at all** in `significant_change`.

`significant_change` _only_ uses `last_service_data`

Reply:
OK perhaps then I don't understand my own code anymore 😂 

Let me re-read how the logic works. 

You're right most likely!

Reply:
No worries! I was in the same boat when I was originally trying to fix #447. I couldn't understand for the life of me what the original code in `significant_change` was doing. So much was there that only existed because you weren't verifying if a transition was taking place. 

To this day after looking of this loop several times I still don't understand the below old code:

```
        for index, old_state in enumerate(old_states):
            changed = compare_to(old_attributes=old_state.attributes)
            if not changed:
                _LOGGER.debug(
                    "State of '%s' didn't change wrt change event nr. %s (context.id=%s)",
                    light,
                    index,
                    context.id,
                )
                break
```
This was driving me nuts as this was the main problem in #447

Reply:
We literally don't need to do anything wacky in significant change other than update the state, and check if the refreshed state matches our last service data.

The ONLY possible cases where `detect_non_ha_changes` can fail in the current version:

1. the bulb never got the light.turn_on command we sent with the last service data (haven't seen this happen in the two weeks I've been using the new `significant_change` code.
2. the bulb went offline during the last adapt, code will currently fire the manual control event for any bulbs that go offline. This was an already outstanding issue iin the old code. (I'll edit back here with the exact issue number)

Reply:
> OK perhaps then I don't understand my own code anymore 😂
> 
> Let me re-read how the logic works.
> 
> You're right most likely!

Hey no worries at all I'm just happy this might actually be implemented to your users as this will close a great many number of old issues!